### ISSUE TITLE
`cloud_cover=100` does not return existing fully cloud-covered scenes

### ISSUE THREAD

**Issue by rafaelfabbrini** [stac-utils/pystac-client#762]
When using the PySTAC client to query datasets with `eo:cloud_cover=100`, no scenes are returned. This behavior contrasts with s similar query on the USGS EarthExplorer, where the same dataset and parameter configuration successfully return fully cloud-covered scenes.

Based on the EO extension documentation, the `eo:cloud_cover` field represents an estimate of cloud cover as a percentage (0–100). While the exclusion of "nodata" areas may influence cloud cover calculations, fully cloud-covered scenes are expected to appear when explicitly requesting `cloud_cover=100`.

---

**Query Setup:**

```python
from pystac_client import Client

catalog_url = "https://landsatlook.usgs.gov/stac-server"
client = Client.open(catalog_url)

search = client.search(
    collections=["landsat-c2l2-sr"],
    query={
        "eo:cloud_cover": {"eq": 100},
        "landsat:collection_category": {"eq": "T1"},
        "landsat:wrs_path": {"eq": "018"},
        "landsat:wrs_row": {"eq": "030"},
    },
    datetime="2012-01-01/2012-05-31",
    max_items=50000,
)

results = list(search.get_items())
print(f"Number of items returned: {len(results)}")
```

**Dataset Example:**
The example above queries the Landsat catalog. On USGS EarthExplorer, the same cloud cover parameter (`eo:cloud_cover=100`) on the same dataset successfully returns fully cloud-covered scenes, which are absent in the PySTAC client results.

**Observed Behavior:**
The PySTAC query returns zero results, even though scenes with full cloud coverage exist and can be retrieved through other platforms.

**Expected Behavior:**
The query should return all items where `eo:cloud_cover` is exactly 100%.

---

**Environment:**

PySTAC version: `0.8.5`
Python version: `3.10.14`
STAC API: `https://landsatlook.usgs.gov/stac-server`

---

**Additional Notes:**

It is unclear whether this behavior stems from:

- A bug in the PySTAC query translation.
- A discrepancy in cloud cover calculation between the data source and PySTAC.
- A misunderstanding of the eo:cloud_cover field's expected behavior in PySTAC queries.

The fact that USGS EarthExplorer successfully retrieves fully cloud-covered scenes under the same conditions suggests the issue may be specific to PySTAC's query processing. Any clarification or fixes would be greatly appreciated. If needed, I can provide further examples or logs.

**Comment by gadomski** [comment]
I don't think this is a **pystac-client** issue. I used the (undocumented) [get_parameters](https://github.com/stac-utils/pystac-client/blob/83dbcd80e1249057adc0df6fb0f06e9e756e175d/pystac_client/item_search.py#L187-L194) method to export the JSON search parameters:

```json
{
    "datetime": "2012-01-01T00:00:00Z/2012-05-31T23:59:59Z",
    "collections": [
        "landsat-c2l2-sr"
    ],
    "query": {
        "eo:cloud_cover": {
            "eq": 100
        },
        "landsat:collection_category": {
            "eq": "T1"
        },
        "landsat:wrs_path": {
            "eq": "018"
        },
        "landsat:wrs_row": {
            "eq": "030"
        }
    }
}
```

I then used **curl** to confirm that the search does not return any results when _not_ using **pystac-client**:

```shell
$ curl --json @issues_762.json https://landsatlook.usgs.gov/stac-server/search
{"type":"FeatureCollection","stac_version":"1.0.0","stac_extensions":[],"context":{"limit":10,"matched":0,"returned":0},"numberMatched":0,"numberReturned":0,"features":[],"links":[]}
```

The same request w/ some tweaks does return items, e.g. if you relax the datetime bounds or change the `eo:cloud_cover` request. 

From your original issue, you say:

> Expected Behavior:
> The query should return all items where eo:cloud_cover is exactly 100%.

That's not what the code that you provide is asking for. The code you provide is asking for all items in a specific row and path within a specific datetime bound and with a specific collection category that has `eo:cloud_cover` equal to 100. That's a much more specific request.

Closing as can't reproduce.

**Comment by hrodmn** [comment]
Hi @rafaelfabbrini :wave: 

As @gadomski mentioned, your STAC API query is very specific. If you relax some of the **other** constraints in your query, you can see that the `eo:cloud_cover` query is working correctly:
```python
from pystac_client import Client

catalog_url = "https://landsatlook.usgs.gov/stac-server"
client = Client.open(catalog_url)

search = client.search(
    collections=["landsat-c2l2-sr"],
    query={
        "eo:cloud_cover": {"eq": 100},
        # "landsat:collection_category": {"eq": "T1"},  # skip this constraint
        "landsat:wrs_path": {"eq": "018"},
        "landsat:wrs_row": {"eq": "030"},
    },
    datetime="2012-01-01/2012-05-31",
    max_items=50000,
)

print("number of matching items:", search.matched())

for item in search.item_collection():
    print(item.id, item.ext.eo.cloud_cover)
```

You
